### PR TITLE
Remove most mentions of type comments from docs

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: docs/source/kinds_of_types.rst [hunk 1]
```diff
@@ -347,23 +347,14 @@ This also works for attributes defined within methods:
         def __init__(self) -> None:
             self.count: Optional[int] = None
 
-As a special case, you can use a non-optional type when initializing an
-attribute to ``None`` inside a class body *and* using a type comment,
-since when using a type comment, an initializer is syntactically required,
-and ``None`` is used as a dummy, placeholder initializer:
```
Comment (AlexWaygood) [inline docs/source/kinds_of_types.rst:350]:
```suggestion
```

File: docs/source/type_inference_and_annotations.rst [hunk 2]
```diff
@@ -44,28 +44,11 @@ type:
...
-.. code-block:: python
-
-   x: str  # Declare type of 'x' without initialization
-
 .. note::
 
    The best way to think about this is that the type annotation sets the
    type of the variable, not the type of the expression. To force the
-   type of an expression you can use :py:func:`cast(\<type\>, \<expression\>) <typing.cast>`.
+   type of an expression you can use :py:func:`cast(\<type\>, \<expression\>) <typing.c#ast>`.
```
Comment (AlexWaygood) [inline docs/source/type_inference_and_annotations.rst:51]:
```suggestion
   type of an expression you can use :py:func:`cast(\<type\>, \<expression\>) <typing.cast>`.
```

Comment (AlexWaygood) [inline docs/source/type_inference_and_annotations.rst:51]:
```suggestion
   type of an expression you can use :py:func:`cast(\<type\>, \<expression\>) <typing.cast>`.
```

Comment (hauntsaninja) [reply docs/source/type_inference_and_annotations.rst]: Sloppy, not sure what happened there, sorry about that

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**